Source/WebCore/platform/graphics/Damage.h

```diff
@@ -250,7 +253,7 @@ class Damage {
             return false;
 
         // When both Damage are already united and have the same rect, we can just iterate the rects and unite them.
-        if (m_mode == Mode::Rectangles && m_shouldUnite && m_rect == other.m_rect && m_shouldUnite == other.m_shouldUnite) {
+        if (m_mode == Mode::Rectangles && m_shouldUnite && m_mode == other.m_mode && m_rect == other.m_rect && m_shouldUnite == other.m_shouldUnite && m_rects.size() == other.m_rects.size()) {
```

Comment:
`m_shouldUnite` would short-circuit if false, so we can take `m_shouldUnite == oher.m_shouldUnite` and leave it as `other.m_shouldUnite`